packages/volto/src/components/manage/Form/BlocksToolbar.jsx

```diff
@@ -97,8 +98,8 @@ export class BlocksToolbarComponent extends React.Component {
         return mode === 'copy'
           ? blockConfig.cloneData
             ? blockConfig.cloneData(blockData)
-            : [uuid(), blockData]
-          : [uuid(), blockData]; // if cut/pasting blocks, we don't clone
+            : [uuid(), cloneBlocks(blockData)]
+          : [uuid(), cloneBlocks(blockData)]; // if cut/pasting blocks, we don't clone
       })
       .filter((info) => !!info); // some blocks may refuse to be copied
```

Comment:
Can we avoid duplicating cloneWithIds between here and cloneBlocks.ts?

Reply:
@davisagli do you mean unify all in a single helper? or improve the condition?

Reply:
@sneridagh I mean use a helper function in both places

Reply:
@davisagli let's talk before you leave, I think that the condition was wrong, from the beginning. If it's not copy, does the same as copy, maybe it's a copy/paste omission error? I'd say if it's move, we should not clone.

Reply:
@sneridagh Ah I see what you are saying.

You can ignore my comment here, I thought this was the same code as in cloneBlocks.ts but it isn't.

But you're right, cloneBlocks should not be used when moving

Reply:
@davisagli okay, I implemented a way to preserve the original uuid if it's a cut operation.